ssz_serialization/types.nim

```diff
@@ -279,6 +281,24 @@ func cacheNodes*(depth, leaves: int): int =
     res += nodesAtLayer(i, depth, leaves)
   res
 
+# Indicates that after growing, the hashes, while they'll be considered cached,
+# also do not allow inference that layers nearer the Merkle hash tree have also
+# been cleared and that the hash cache clearing must continue to the root.
+#
+# False positives are possible, but harmless: if a cleared entry coincidentally
+# matches this pattern, then it inefficiently recomputes some Merkle tree nodes
+# and still creates a correct result.
+const uninitSentinel = Digest(data: [
+  0, 0, 0, 0, 0, 0, 0, 0,
+  1, 1, 1, 1, 1, 1, 1, 1,
+  1, 1, 1, 1, 1, 1, 1, 1,
+  1, 1, 1, 1, 1, 1, 1, 1])
+
+# Aside from these conditions, the specific value of the sentinel is arbitrary.
+static:
+  doAssert uninitSentinel != default(Digest)
+  doAssert uninitSentinel.data.toOpenArray(0, 7) == default(array[8, byte])
```

Comment:
```suggestion
  doAssert not isCached(uninitSentinel.data)
```

Reply:
Tried that first, but it takes an `addr`, which Nim doesn't like in the VM.

I'm not sure this is a Nim bug, just a quirk of using the VM.

Reply:
But yes, I'd prefer that too.

Reply:
we can put `when nimvm` in `isCached` so we keep the magic value checking in one place

Reply:
> we can put `when nimvm` in `isCached` so we keep the magic value checking in one place

Okay, this avoids the need to making sure that benchmarkingwise it doesn't case other unrelated regressions, so it's safe.

Reply:
https://github.com/status-im/nim-ssz-serialization/pull/66/commits/cf2319bdff5c63d04b320f830dd7ea8b7e152bc6